algotaurus.py

```diff
@@ -769,17 +883,94 @@ def execute_code(self):
             self.buttstop.configure(state='disabled')
             self.buttstep.configure(state='normal')
             self.buttrun.configure(state='normal')
             self.textPad.configure(state='normal', bg='white')
             self.execute = False
 
+class AlgoTaurusMessageBox():
+
+    def __init__(self, parent=None, title=None, message=None, icon=None):
+        import Tkinter as tk
+        import ttk
+        from PIL import ImageTk, Image
+        self.tk = tk
+        self.ttk = ttk
+        self.ImageTk = ImageTk
+        self.Image = Image
+
+        self.var = True
+        self.parent = parent
+        self.window = tk.Toplevel(parent)
+        self.message = message
+        self.title = title
+        self.image = ImageTk.PhotoImage(self.Image.open(icon).resize((100, 100), Image.ANTIALIAS))
+        self.window.attributes("-toolwindow",1)
+        self.window.wm_title(self.title)
+        self.window.resizable(0,0)
+
+    def ok_cmd(self, event=None):
+        self.window.destroy()
+        self.var = True
+
+    def cancel_cmd(self, event=None):
+        self.window.destroy()
+        self.var = False
+
+    def draw_message(self, buttons=1):
+        f1 = self.tk.Frame(self.window, bg='white')
+        f2 = self.tk.Frame(self.window)
+        f1.pack()
+        f2.pack(fill='x')
+        imglabel = self.tk.Label(f1, image = self.image, bg='white')
+        imglabel.image = self.image
+        imglabel.grid(row=0, column=0, padx = 20, pady = 20, sticky='e')
+        self.tk.Label(f1, text=self.message, font=("Helvetia", 12), bg='white', justify='left').grid(row=0, column=1, padx = 10)
+        ok_side = 'right'
+        if buttons > 1:
+            btn_cancel = self.ttk.Button(f2, text=_('Cancel'), command=self.cancel_cmd)
+            btn_cancel.bind('<Return>', self.cancel_cmd)
+            btn_cancel.pack(side = 'right', padx=10, pady=10)
+            btn_cancel.focus_set()
+            ok_side = 'left'
+        btn_ok = self.ttk.Button(f2, text='OK', command=self.ok_cmd)
+        btn_ok.bind('<Return>', self.ok_cmd)
+        btn_ok.pack(side=ok_side, padx=10, pady=10)
+        if buttons == 1:
+            btn_ok.focus_set()
+        else:
+            self.window.bind('<Left>', lambda event: btn_ok.focus_set())
+            self.window.bind('<Right>', lambda event: btn_cancel.focus_set())
+        self.window.bind('<Escape>', self.cancel_cmd)
+        x = (self.window.winfo_screenwidth() - self.window.winfo_reqwidth()) / 2
+        y = (self.window.winfo_screenheight() - self.window.winfo_reqheight()) / 2
+        self.window.geometry("+%d+%d" % (x-150, y))
+        self.window.focus()
+        self.window.grab_set()
+        self.window.wait_window()
+        return self.var
+
+    def showinfo(self):
+        return self.draw_message(1)
+
+    def askokcancel(self):
+        return self.draw_message(2)            
```

Comment:
Class of the new, user-friendly messageboxes. It requies a title, a message and an icon file to display.